src/Xamarin.Android.Tools.Bytecode/Kotlin/KotlinUtilities.cs

```diff
@@ -72,6 +72,29 @@ public static string GetMethodNameWithoutSuffix (this MethodInfo method)
 			return index >= 0 ? method.Name.Substring (0, index) : method.Name;
 		}
 
+		public static bool IsDefaultConstructorMarker (this MethodInfo method)
+		{
+			// A default constructor is synthetic and always has an int and a
+			// DefaultConstructorMarker as its final 2 parameters.
+			if (method.Name != "<init>")
+				return false;
+
+			if (!method.AccessFlags.HasFlag (MethodAccessFlags.Synthetic))
+				return false;
+
+			var parameters = method.GetParameters ();
+
+			if (parameters.Length < 2)
+				return false;
+
+			// Final parameter is DefaultConstructorMarker
+			if (parameters.Last ().Type.TypeSignature != "Lkotlin/jvm/internal/DefaultConstructorMarker;")
```

Comment:
`.Last()` involves an IEnumerable allocation & enumeration.  `parameters [parameters.Length-1]` would be more efficient, and in keeping with the `return` statement.